tests/test_benchmarks.py

```diff
@@ -0,0 +1,433 @@
+            "-q",
+            "--format",
+            "pretty",
+            tmp_path / "benchmark.json",
+            exit_code=0,
+        )
+
+        # FIXME: We don't assert against the exact output yet, as it's a WIP
+        assert stdout, stderr
+
+    @pytest.mark.asyncio
+    async def test_benchmark_run_markdown_output(
+        self,
+        valid_single_benchmark,
+        tmp_path,
+    ):
+        tmp_path.joinpath("benchmark.json").write_text(
+            json.dumps(valid_single_benchmark),
+        )
+        stdout, stderr = await bowtie(
+            "perf",
+            "-i",
+            DIRECT_CONNECTABLE,
+            "-q",
+            "--format",
+            "markdown",
+            tmp_path / "benchmark.json",
+            exit_code=0,
+        )
+
+        expected_data1 = """
+# Benchmark Summary
+## Benchmark Group: benchmark
+Benchmark File: stdin
+Benchmark: benchmark
+
+| Test Name | python-jsonschema |
+        """.strip()
+
+        expected_data2 = """
+## Benchmark Metadata
+
+Runs: 3
+Values: 2
+Warmups: 1
+        """.strip()
+
+        # Cant verify the whole output as it would be dynamic
+        # with differing values
+        assert expected_data1 in stdout
+        assert expected_data2 in stdout
+
+    @pytest.mark.asyncio
+    async def test_invalid_benchmark_run(self, invalid_benchmark, tmp_path):
+        tmp_path.joinpath("benchmark.json").write_text(
+            json.dumps(invalid_benchmark),
+        )
+        _, stderr = await bowtie(
+            "perf",
+            "-i",
+            DIRECT_CONNECTABLE,
+            "-q",
+            "--format",
+            "json",
+            tmp_path / "benchmark.json",
+            exit_code=EX.DATAERR,
+        )
+        assert "benchmark-load-error" in stderr
+
+
+class TestFilterBenchmarks:
+
+    @pytest.mark.parametrize(
+        "dialect",
+        Dialect.known(),
+        ids=lambda param: param.short_name,
+    )
+    @pytest.mark.asyncio
+    async def test_default_benchmarks(self, dialect):
+        stdout, stderr = await bowtie(
+            "filter-benchmarks",
+            "-D",
+            dialect.short_name,
+            exit_code=0,
+        )
+        assert stderr == ""
+
+    @pytest.mark.parametrize(
+        "dialect",
+        Dialect.known(),
+        ids=lambda param: param.short_name,
+    )
+    @pytest.mark.asyncio
+    async def test_keyword_benchmarks(self, dialect):
+        stdout, stderr = await bowtie(
+            "filter-benchmarks",
+            "-D",
+            dialect.short_name,
+            "-t",
+            "keyword",
+            exit_code=0,
+        )
+        assert stderr == ""
+
+    @pytest.mark.asyncio
+    async def test_filtering_by_name(self):
+        stdout, stderr = await bowtie(
+            "filter-benchmarks",
+            "-t",
+            "keyword",
+            "-n",
+            uuid.uuid4().hex,
+            exit_code=0,
+        )
+        assert stdout == "", stderr == ""
```

Comment:
This assert probably doesn't do what you expect, the syntax is `assert foo, message` where the latter is a message shown on failure (so here you'd show a message of `True` or `False` but not assert against it). Are you trying to assert 2 things here presumably?

Reply:
Oh my bad, I thought it did the same